src/framework/theme/components/select/select.component.ts

```diff
@@ -371,6 +369,7 @@ export class NbSelectComponent<T> implements OnInit, AfterViewInit, AfterContent
 
   writeValue(value: T | T[]): void {
     if (!value) {
+      this.cleanSelection();
```

Comment:
Can't we keep `OnPush` and manually call `this.cd.detectChanges` here after selection cleanup?